services/provider/server/server.go

```diff
@@ -596,5 +526,119 @@ func (s *OCSProviderServer) RevokeStorageClassClaim(ctx context.Context, req *pb
 
 // GetStorageClassClaim RPC call to get the ceph resources for the StorageclassClaim.
 func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req *pb.StorageClassClaimConfigRequest) (*pb.StorageClassClaimConfigResponse, error) {
-	return &pb.StorageClassClaimConfigResponse{}, nil
+	consumerClaimObj, err := s.storageClassClaimManager.Get(ctx, req.StorageConsumerUUID, req.StorageClassClaimName)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+		}
+		return nil, status.Errorf(codes.Internal, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+	}
+
+	// TODO: add status.Phase validation here to check claim is in expected
+	// state or not before extracting the ceph resources from the StorageClassClaim CR status
```

Comment:
Are we waiting for something? Can't we just check for `consumerClaimObj.Status.Phase == "Ready"`?

Reply:
Yes i can do it but below is the reason for not doing it

* Still types (status.Phases) are not defined in the types.go as it need to be added by the server
* Multiple checks need to be added to return the proper GRPC error code.

Let me know if you still want to just add string comparison and keep the TODO and replace it when consts are defined in the storageclassclaim_types.go file.

Reply:
So this is going to be implemented in a future PR, correct?

Reply:
Yes correct it will be in the future once the implementation is done for storageclaim controller on provider side.